buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -610,6 +610,17 @@ class ClusterFormationTasks {
                 waitFailed(project, nodes, logger, 'Failed to start elasticsearch')
             }
 
+            // make sure all files exist otherwise we haven't fully started up
+            boolean missingFile = false
+            for (NodeInfo node : nodes) {
+                missingFile |= node.pidFile.exists() == false
+                missingFile |= node.httpPortsFile.exists() == false
+                missingFile |= node.transportPortsFile.exists() == false
+            }
+            if (missingFile) {
```

Comment:
What do you think of making this a set that we remove from and spill if it's not empty at the end? Then we can give indication of what is missing?

Reply:
I'm good with that but to be clear we will be duplicating data that is already printed out. `waitFailed` prints the following lines:

```java
logger.error("|  failure marker exists: ${node.failedMarker.exists()}")
logger.error("|  pid file exists: ${node.pidFile.exists()}")
logger.error("|  http ports file exists: ${node.httpPortsFile.exists()}")
logger.error("|  transport ports file exists: ${node.transportPortsFile.exists()}")
```

Reply:
Oh, that's fine then.